UnattendResources/Logon.ps1

```diff
@@ -75,9 +75,16 @@ function Clean-UpdateResources {
     Remove-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" -Name AutoLogonCount -ErrorAction SilentlyContinue
 
     # Cleanup
-    Remove-Item -Recurse -Force $resourcesDir
+    Remove-Item -Recurse -Force -Exclude "zapfree.exe" $resourcesDir
     Remove-Item -Force "$ENV:SystemDrive\Unattend.xml"
-

```

Comment:
I think the placement of zapfree execution should not be here, as it is optimistically at best.
At this stage, sysprep is already executing and sysprep might finish and shutdown the VM before zapfree finishes, leaving a possible broken image or a huge file created by zapfree.

Maybe https://github.com/cloudbase/windows-openstack-imaging-tools/pull/261/files#diff-b0340988ad4db4e9c00077d542df3922R401 (putting zapfree right before the sysprep) makes more sense.

Reply:
To give more insight, sysprep is executed async mode, and it will turn off the vm after it finishes.

Reply:
This raises the question:
Why are there any actions *after* sysprep at all if they are not guaranteed to finish?

Reply:
A sysprep basic run takes around several minutes, while the unattend resources cleanup takes a few seconds. But you have a point there. Is a little of cat and mouse problem, as I cannot remove the unattend resource dir before as it contains the unattend.xml file or some Run-CustomScript "RunAfterSysprep.ps1" hook required files. The best way is to remove the /shutdown flag to sysprep, then add a wait loop until syspre.exe finishes (by checking that the process exists) and then perform cleanup/zap + the manual shutdown.

Reply:
Well, that's a possible topic for a new PR. For now, I just moved the zap invocation before the sysprep invocation.